File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
cut = input()
cnt = 0
iron = 0
i = 0
while i <len(cut):
    if cut[i] == '(' and cut[i+1] == ')':
        cnt += iron
        i += 2 #레이저크기만큼
    elif cut[i] == '(':
        iron +=1
        i+=1
    elif cut[i] == ')':
        iron -=1
        cnt +=1 #쇠막대기 길이가 끝나 +1
        i+=1
print(cnt)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <string>
using namespace std;

int main(void)
{
	char cut[100001] = { 0, };
	cin >> cut;
	int iron = 0;
	int ir = 0;//raser
	int cnt = 0;
	while (cut[ir] != 0)
	{
		if (cut[ir] == '(' && cut[ir+1] == ')')
		{
			cnt += iron;
			ir += 2;
		}
		else if (cut[ir] == '(')
		{
			iron += 1;
			ir += 1;
		}
		else if (cut[ir] == ')')
		{
			iron -= 1;
			cnt += 1; //stick end
			ir += 1;
		}
	}
	cout << cnt << '\n';
	return 0;
}